### DISCUSSION TITLE
Stuck in preparing Nix Flakes

### DISCUSSION THREAD

**Discussion by Anifyuli** [nix-community/nix-on-droid#251] [Q&A]
Hello, Mister. I want to report my problem in post-install Nix on Droid on my device. It problem is stucking on `evaluating derivation 'path:/data/data/...'`. I try to erasing data and trying again but I encountered with this problem. What should I do, Mister? Can this problem solved? Thanks for read my issues 

Device info :
SoC : Qualcomm Snapdragon 720G
Device : OPPO Reno5 (4G edition)
OS : ColorOS 12 based on Android 12 stable release 
Mem/Storage : 8/128 GB

![Screenshot_2023-02-18-06-07-28-69_7c9fdf2472f91e6d21f35bb801178e75.jpg](https://user-images.githubusercontent.com/73734843/219814241-1bd4c979-a0c3-4257-af51-d453e88a0c68.jpg)

**Comment by Gerschtli** [comment]
Probably just wait, this initial setup is really depending on your device. Like 30-45 minutes on my oneplus 5. In your android settings have a look at the growing size of used storage for the Nix-on-Droid app.

**Comment by Anifyuli** [comment]
> Probably just wait, this initial setup is really depending on your device. Like 30-45 minutes on my oneplus 5. In your android settings have a look at the growing size of used storage for the Nix-on-Droid app.

I waiting for 12 hours but Nix on Droid not change to other prep steps

**Comment by Gerschtli** [comment]
Oh okay, that sounds wrong. Can you check if the storage size increased? Just to make sure, it downloads anything. Maybe your internet speed is slow but tbh I don't what could cause that kind of long build times.

The flake setup is usually a bit slower than the channel one, you could try if that is making a difference but otherwise I am out of ideas.. @t184256 ?

**Comment by t184256** [comment]
It's not gonna grow during evaluation,  is it?

Two ideas:

1. to build a bootstrap tarball with `-vvv` in the right place and debug further, but it'll show where it's stuck and not why it's stuck and likely won't help us
2. if it's some super-aggressive power management at play, try to follow the steps from https://dontkillmyapp.com, then restart the installation, acquire wakelock through the notification and see whether that helps

**Comment by Gerschtli** [comment]
You are probably right..

**Comment by Anifyuli** [comment]
After acquire wakelock and enabling don't disable background process for Nix on Droid setup operations can done

![Screenshot_2023-02-18-20-22-27-00_7c9fdf2472f91e6d21f35bb801178e75.jpg](https://user-images.githubusercontent.com/73734843/219868402-394790b9-eb83-4f20-85c0-fff4e15b954a.jpg)
